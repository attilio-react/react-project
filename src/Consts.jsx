const consts = { 
    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1
}

export {consts}
