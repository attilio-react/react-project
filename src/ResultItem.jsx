import React from 'react';

import {Label} from './Label.jsx';
import {Button} from './Button.jsx';

class ResultItem extends React.PureComponent {
  render () {
    const {item} = this.props;
    return <div>
                  <p><img src={item.imgSrc} /></p>
                  <p>
		     <Label text={item.releaseDate} />
		     &nbsp;
                     <Label text={item.genre} />
		  </p>
                  <p><Label text={item.title} /></p>
                  
           </div>;
  }
}


export {SearchByContainer}


