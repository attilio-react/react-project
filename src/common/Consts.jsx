const consts = { 
    SEARCH_SCREEN : 0,
    DETAIL_SCREEN: 1,


    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1,


    PARAM_SORT_BY: 'sortBy',
    PARAM_SORT_ORDER: 'sortOrder',
    PARAM_SEARCH: 'search',
    PARAM_SEARCH_BY: 'searchBy',

    VALUE_BY_GENRE: 'genres',
    VALUE_BY_TITLE: 'title',
    VALUE_BY_RELEASE_DATE: 'release_date',
    VALUE_BY_VOTE_AVERAGE: 'vote_average',

    VALUE_ASC: 'asc',
    VALUE_DESC: 'desc'
}

export {consts}
