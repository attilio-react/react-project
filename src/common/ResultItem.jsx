import React from 'react';
import {Link} from "react-router-dom";

import {paths} from 'Common/Paths.jsx';

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class ResultItem extends React.PureComponent {

 render () {
    const {item, clickCallback} = this.props,
          self = this;
    return <div style={{border: '1px solid black'}}><Link to={paths.DETAILS_PATH + item.id}>
                  <p><img src={item.poster_path} /></p>
                  <p>
		     <Label text={item.release_date} />
		     &nbsp;
                     <Label text={item.genres.join(', ')} />
		  </p>
                  <p><Label text={item.title} /></p>
                 </Link> 
           </div>;
  }
}


export {ResultItem}


