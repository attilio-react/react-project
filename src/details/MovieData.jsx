import React from 'react';

import {DetailsContext} from './DetailsContext.jsx'

class MovieData extends React.PureComponent {
  render () {
    return (<DetailsContext.Consumer>
            {ctx => (
                <div>
                    <img src={ctx.movie.poster_path} />
                    <p><span>{ctx.movie.title}</span></p>
                    <p><span>{ctx.movie.vote_average}</span></p>
                    <p><span>{ctx.movie.release_date}</span></p>
                    <p><span>{ctx.movie.runtime}</span></p>
                    <p><span>{ctx.movie.overview}</span></p>
                </div>
            )}
            </DetailsContext.Consumer>)
  }

}


export {MovieData}


