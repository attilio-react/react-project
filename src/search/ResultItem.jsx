import React from 'react';

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class ResultItem extends React.PureComponent {
  render () {
    const {item} = this.props;
    return <div>
                  <p><img src={item.poster_path} /></p>
                  <p>
		     <Label text={item.release_date} />
		     &nbsp;
                     <Label text={item.genres.join(', ')} />
		  </p>
                  <p><Label text={item.title} /></p>
                  
           </div>;
  }
}


export {ResultItem}


