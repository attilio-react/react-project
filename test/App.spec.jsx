import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';


import {App} from '|App.jsx';
import {ResultItem} from '|common/ResultItem.jsx';
import {MovieData} from '|details/MovieData.jsx';
import {consts} from '|common/Consts.jsx';

describe('Rendering', () => {
  it('Should render without results', () => {
    const 
        component = renderer.create(<App />),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Should render with results', (done) => {
    const 
        component = mount(<App />)
    component.setState({
        total: 10,
        items: [
	    {
		id: 1,
		poster_path: 'http://hello.com/poster.png',
		title: 'Movie Title',
		vote_average: 4.4,
		release_date: '2011-01-01',
		runtime: 129,
		overview: 'This is a movie about ....',
		genres: ['Horror', 'Thriller']
	    },
	    {
		id: 2,
		poster_path: 'http://hello.com/poster2.png',
		title: 'Another Movie Title',
		vote_average: 4.4,
		release_date: '2012-02-02',
		runtime: 139,
		overview: 'This is another movie about ....',
		genres: ['War', 'History']
	    }
	]
    }, () => {
        expect(component.find(ResultItem).length).toBe(2);
        done()
    })
  });

  it('Should render with details', (done) => {
    const 
        component = mount(<App />)
    component.setState({
        screen: consts.DETAIL_SCREEN,
        selectedMovie: 
	    {
		id: 1,
		poster_path: 'http://hello.com/poster.png',
		title: 'Movie Title',
		vote_average: 4.4,
		release_date: '2011-01-01',
		runtime: 129,
		overview: 'This is a movie about ....',
		genres: ['Horror', 'Thriller']
	    },
        sameGenreMovies: [
	    {
		id: 2,
		poster_path: 'http://hello.com/poster2.png',
		title: 'Another Movie Title',
		vote_average: 4.4,
		release_date: '2012-02-02',
		runtime: 139,
		overview: 'This is another movie about ....',
		genres: ['War', 'History', 'Horror']
	    },
	    {
		id: 3,
		poster_path: 'http://hello.com/poster3.png',
		title: 'Yet Another Movie Title',
		vote_average: 4.5,
		release_date: '2012-03-02',
		runtime: 129,
		overview: 'This is yet another movie about ....',
		genres: ['Horror', 'History']
	    },
	    {
		id: 4,
		poster_path: 'http://hello.com/poster4.png',
		title: 'And Yet Another Movie Title',
		vote_average: 3.5,
		release_date: '2013-03-02',
		runtime: 127,
		overview: 'This is yet another movie about ....',
		genres: ['Horror', 'History', 'War']
	    }
        ],
        selectedMovieGenres: ['Horror', 'Thriller']

    }, () => {
        expect(component.find(ResultItem).length).toBe(3);
        expect(component.find(MovieData).length).toBe(1);
        done()
    })
  });
});


describe('Actions', () => {
  it('Should be able to search for movies', (done) => {
    const 
        component = mount(<App />),
        RESPONSE = {"data":[{"id":1,"title":"Title","tagline":"Tagline","vote_average":6.1,"vote_count":1195,"release_date":"2018-02-07","poster_path":"http://path/to/poster.jpg","overview":"OVERVIEW","budget":1,"revenue":2,"genres":["Drama"],"runtime":3}, {"id":2,"title":"Title2","tagline":"Tagline2","vote_average":6.2,"vote_count":1198,"release_date":"2018-02-08","poster_path":"http://path/to/poster2.jpg","overview":"OVERVIEW2","budget":2,"revenue":3,"genres":["Thriller"],"runtime":4}],"total":2,"offset":0,"limit":10}

    fetch.mockResponseOnce(JSON.stringify(RESPONSE))

    component.setState({
        searchTerm: 'movie title'
    }, () => {
        component.instance().searchMovies(() => {
            component.update()
            expect(component.find(ResultItem).length).toBe(2);
            done()
        })
    })
  });
})
