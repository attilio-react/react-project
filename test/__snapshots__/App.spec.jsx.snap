// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResult rendering Should render without results 1`] = `
Array [
  <span>
    netflixroulette
  </span>,
  <p>
    <span>
      FIND YOUR MOVIE
    </span>
  </p>,
  <p>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </p>,
  <p>
    <span>
      SEARCH BY
    </span>
     
    <input
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "underline",
        }
      }
      type="button"
      value="TITLE"
    />
     
    <input
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      type="button"
      value="GENRE"
    />
           
    <input
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      type="button"
      value="SEARCH"
    />
  </p>,
  <div
    style={
      Object {
        "fontSize": "large",
      }
    }
  >
    No films found
  </div>,
  <span>
    netflixroulette
  </span>,
]
`;
